os/os.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package os provides gNOI os operations.
+package os
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	log "github.com/golang/glog"
+	ospb "github.com/openconfig/gnoi/os"
+	"github.com/openconfig/gnoigo/internal"
+)
+
+// InstallOperation represents the parameters of a Install operation.
+type InstallOperation struct {
+	version string
```

Comment:
To make this more consistent with the other operations, maybe let's replace version and standby fields with a single InstallRequest field that gets populated by the Version and Standby functions. This would then become the initial install request that is sent.

Reply:
What do you think about this idea?

Reply:
I was looking at the [installRequest](https://github.com/openconfig/gnoi/blob/main/os/os.proto#L143-L149), and we are only taking 2 fields of `TransferRequest` as the input. It looks simpler to directly use these fields instead of populating the InstallRequest here.

Is the benefit of  having `InstallRequest` field, that , if Version() and Standby() aren't called then TransferRequest message is not set? Instead of the current scenario where Install message will be instantiated with TransferRequest{version: "", StandbySupervisor:false} ??

Reply:
Thinking about this some more and had a couple of questions:

If request is not set in InstallRequest message, is this considered an invalid request? And should `Execute` return an error ?  Or does the device return an error (which would be forwarded by Execute)? 

If yes, it seems that having a field `InstallRequest` that gets populated by the Version() and Standby() functions would be the correct way to set `InstallRequest`. 

So if Version and Standby are not called by the user, then the request would be `InstallRequest{}` without setting the Request field , instead of the following where Request field is set with Version empty string and standby set to false.

```
InstallRequest{
  Request: &ospb.InstallRequest_TransferRequest{
    TransferRequest: &ospb.TransferRequest{
      Version:          "",
      StandbySupervisor: false,
    },
  },
}
```

And if not setting the request field is an invalid request, a error can be returned to user.

Reply:
If request is not set, that would be an invalid request, but `gnoigo` should prevent that from ever happening. That is, `NewInstallOperation` would initialize the req field with `&InstallRequest{Request: &ospb.InstallRequest_TransferRequest{TransferRequest: &ospb.TransferRequest{}}}`.

There isn't _tremendous_ value to doing this, but the arguable value would be:

1. Style consistency between how we code the structs for reach operation. Namely, that we always include a `req` field with the initial request field, perhaps plus some additional fields if needed (probably only for operations involving client-side streaming).
2. If `TransferRequest` were to grow to add additional fields in the future, we wouldn't need to change the `NewInstallOperation` constructor.

I'm mostly focused on point #1, for style/consistency, so those of us coding these things follows the same pattern and everyone reading knows what to expect when they are reading an operation.

One downside is that there are multiple `InstallRequests` over the lifetime of an install operation, so storing _one_ may be a bit misleading? Another option is to store the `TransferRequest` in the field. I could see an argument for that here.

Thoughts?

Reply:
I prefer storing the `TransferRequest` since the InstallOperation will only be storing the initial `InstallRequest` and that would always be `TransferRequest`. Storing `InstallRequest` message would be misleading. 

Also, regarding point 1, currently, Reboot operation also does not store req, will update that in a seperate PR  https://github.com/openconfig/gnoigo/blob/main/system/system.go#L157-L164

Reply:
+1 to including the TransferRequest and +1 to refactoring the Reboot operation to follow the same style. I think this will help create consistency and predictability of implementation.

Reply:
Done and Reboot PR - https://github.com/openconfig/gnoigo/pull/14